homeassistant/components/spotify/media_player.py

```diff
@@ -73,98 +67,50 @@
-        DEFAULT_NAME,
-        lambda _: None,
-        link_name=CONFIGURATOR_LINK_NAME,
-        link_url=oauth.get_authorize_url(),
-        description=CONFIGURATOR_DESCRIPTION,
-        submit_caption=CONFIGURATOR_SUBMIT_CAPTION,
-    )
-
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Spotify platform."""
+    # def play_playlist_service(service):
+    #     media_content_id = service.data[ATTR_MEDIA_CONTENT_ID]
+    #     random_song = service.data.get(ATTR_RANDOM_SONG)
+    #     player.play_playlist(media_content_id, random_song)
 
-    callback_url = f"{hass.config.api.base_url}{AUTH_CALLBACK_PATH}"
-    cache = config.get(CONF_CACHE_PATH, hass.config.path(DEFAULT_CACHE_PATH))
-    oauth = spotipy.oauth2.SpotifyOAuth(
-        config.get(CONF_CLIENT_ID),
-        config.get(CONF_CLIENT_SECRET),
-        callback_url,
-        scope=SCOPE,
-        cache_path=cache,
-    )
-    token_info = oauth.get_cached_token()
-    if not token_info:
-        _LOGGER.info("no token; requesting authorization")
-        hass.http.register_view(SpotifyAuthCallbackView(config, add_entities, oauth))
-        request_configuration(hass, config, add_entities, oauth)
-        return
-    if hass.data.get(DOMAIN):
-        configurator = hass.components.configurator
-        configurator.request_done(hass.data.get(DOMAIN))
-        del hass.data[DOMAIN]
-    player = SpotifyMediaPlayer(
-        oauth, config.get(CONF_NAME, DEFAULT_NAME), config[CONF_ALIASES]
-    )
-    add_entities([player], True)
-
-    def play_playlist_service(service):
-        media_content_id = service.data[ATTR_MEDIA_CONTENT_ID]
-        random_song = service.data.get(ATTR_RANDOM_SONG)
-        player.play_playlist(media_content_id, random_song)
-
-    hass.services.register(
-        DOMAIN,
-        SERVICE_PLAY_PLAYLIST,
-        play_playlist_service,
-        schema=PLAY_PLAYLIST_SCHEMA,
-    )
+    # hass.services.register(
+    #     DOMAIN,
+    #     SERVICE_PLAY_PLAYLIST,
+    #     play_playlist_service,
+    #     schema=PLAY_PLAYLIST_SCHEMA,
+    # )
 
 
-class SpotifyAuthCallbackView(HomeAssistantView):
-    """Spotify Authorization Callback View."""
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], bool], None],
+) -> None:
+    """Set up Spotify based on a config entry."""
+    implementation = await async_get_config_entry_implementation(hass, entry)
 
-    requires_auth = False
-    url = AUTH_CALLBACK_PATH
-    name = AUTH_CALLBACK_NAME
+    session = OAuth2Session(hass, entry, implementation)
 
-    def __init__(self, config, add_entities, oauth):
-        """Initialize."""
-        self.config = config
-        self.add_entities = add_entities
-        self.oauth = oauth
+    player = SpotifyMediaPlayer(
+        session,
+        entry.data[CONF_ID],
+        entry.data[CONF_NAME],
+        hass.data[DOMAIN][CONF_ALIASES],
```

Comment:
This will not exist if there is no config in `configuration.yaml`, because you return True prior to creating this key.